File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeRequestOnTaxCollectorMessage.scala





















// Generated on 05/03/2013 10:34:55
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeRequestOnTaxCollectorMessage extends MessageDeserializer {
  val messageId = 5779

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeRequestOnTaxCollectorMessage.messageId

    val taxCollectorId: Int

    def serialize(buf: Buffer) {
      buf.writeInt(taxCollectorId)
    }

  }

  protected def create(taxCollectorId: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val taxCollectorId = buf.readInt()

    create(taxCollectorId)
  }

}

object ExchangeRequestOnTaxCollectorMessage extends ExchangeRequestOnTaxCollectorMessage {
  case class MessageImpl(taxCollectorId: Int) extends Message
  type Target = MessageImpl

  override def create(taxCollectorId: Int) = MessageImpl(taxCollectorId)

  def apply(taxCollectorId: Int) = create(taxCollectorId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeRequestOnTaxCollectorMessage.java


// Generated on 05/08/2013 19:37:55
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeRequestOnTaxCollectorMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5779;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int taxCollectorId;
    
    public ExchangeRequestOnTaxCollectorMessage() { }
    
    public ExchangeRequestOnTaxCollectorMessage(int taxCollectorId) {
        this.taxCollectorId = taxCollectorId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(taxCollectorId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        taxCollectorId = buf.readInt();
    }
    
}
